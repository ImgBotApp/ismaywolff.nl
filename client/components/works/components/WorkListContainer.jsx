import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { getWorkState } from '../selectors'
import { fetchWorks } from '../actions'
import WorkList from './WorkList'

export class WorkListContainer extends Component {
  componentDidMount() {
    if (this.props.works.result.length === 0) {
      this.props.fetchWorks()
    }
  }

  render() {
    const { works } = this.props
    return <WorkList ids={works.result} isFetching={works.isFetching} />
  }
}

WorkListContainer.propTypes = {
  fetchWorks: PropTypes.func.isRequired,
  works: PropTypes.shape({
    isFetching: PropTypes.bool.isRequired,
    result: PropTypes.array.isRequired
  }).isRequired
}

const mapStateToProps = state => ({
  works: getWorkState(state)
})

export default connect(mapStateToProps, { fetchWorks })(WorkListContainer)
