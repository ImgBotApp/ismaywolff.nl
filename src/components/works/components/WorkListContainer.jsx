import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { fetchWorks } from '../actions'
import WorkList from './WorkList'

export class WorkListContainer extends Component {
  componentDidMount() {
    this.props.fetchWorks()
  }

  render() {
    const { isFetching, result } = this.props.works
    return <WorkList ids={result} isFetching={isFetching} />
  }
}

WorkListContainer.propTypes = {
  fetchWorks: PropTypes.func.isRequired,
  works: PropTypes.object.isRequired
}

const mapStateToProps = state => ({
  works: state.works
})

export default connect(mapStateToProps, { fetchWorks })(WorkListContainer)
