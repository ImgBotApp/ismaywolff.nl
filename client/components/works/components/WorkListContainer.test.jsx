import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { WorkListContainer } from './WorkListContainer'

describe('<WorkListContainer />', () => {
  it('renders correctly', () => {
    const works = { result: ['id'], isFetching: false }
    const wrapper = shallow(<WorkListContainer works={works} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
