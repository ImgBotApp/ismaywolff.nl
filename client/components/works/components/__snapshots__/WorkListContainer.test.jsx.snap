// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkListContainer /> renders correctly 1`] = `
<WorkList
  ids={
    Array [
      "id",
    ]
  }
  isFetching={false}
/>
`;
